File: cpp/sem22/linear_search.cpp
#include <iostream>
using namespace std;

class linear_search
{
public:
    int i, n, item, arr[50];
    void input();
    void search();
};

void linear_search::input()
{
    cout << "Enter array limit: ";
    cin >> n;
    cout << "Enter array elements: " << endl;
    for (i = 1; i <= n; i++)
    {
        cin >> arr[i];
    }
    cout << "Enter the search item: ";
    cin >> item;
}

void linear_search::search()
{
    for (i = 1; i <= n; i++)
    {
        if (arr[i] == item)
        {
            cout << item << " is found at Loction : " << i;
            break;
        }
    }
    if (i > n)
    {
        cout << item << " is not found in the array.";
    }
}

int main()
{
    linear_search obj;
    obj.input();
    obj.search();

    return 0;
}

File: python/problems/linear_search.py

lista = []

n = input("Enter the list limit: ")
n = int(n)
print("Enter the list elements: \n")


def input_fun():
    for i in range(n):
        element = int(input())
        lista.append(element)
    print(lista)


def linear_search(list1, key):
    key = int(input("Enter the key element for search: "))
    i = 0
    flag = False
    while i < len(list1) and not flag:
        if list1[i] == key:
            flag = True
            print("The key element is found at" + str(list[i] + " Location"))
        else:
            i = i + 1
    return flag


input_fun()

linear_search(lista, key)
